src/frequency.rs

```diff
@@ -111,9 +111,7 @@ impl Rate {
 
     /// Multiplication against u32 with overflow check
     pub fn checked_mul(self, other: u32) -> Option<Self> {
-        self.hz.checked_mul(other).map(
-            |r| Self { hz: r }
-        )
+        self.hz.checked_mul(other).map(Self::hz)
```

Comment:
I don't understand why `Self::hz` is a function there

Reply:
@martinber `.map(Self::hz)` is the same thing, and prefered way, to do `.map(|r| Self::hz(r))`. And `Self` in this place is same as `Rate`.

I prefer as much as possible to call the implemented functions (like `Rate::hz()`) instead of constructing structs by hand (`Rate { hz: r }`). This way whatever future validation, changes, etc. are done to `Rate::hz()` are already used in other places constructing it.

Compiler should be intelligent enough to inline the function call so there should be no performance issues either.

Reply:
Oh, yes, I forgot that there was a `Rate::hz(u32)` function. I thought that somehow `Rate::hz` was referring to the `hz` field inside the struct (defined as `struct Rate { hz: u32 }`)